metaflow/plugins/kfp/tests/run_integration_tests.py

```diff
@@ -43,18 +49,67 @@ def _python():
         return "python"
 
 
+non_standard_test_flows = [
+    "raise_error_flow.py",
+    "accelerator_flow.py",
+    "s3_sensor_flow.py",
+    "upload_to_s3_flow.py",
+]
+
+
 def obtain_flow_file_paths(flow_dir_path: str) -> List[str]:
     file_paths = [
         file_name
         for file_name in listdir(flow_dir_path)
         if isfile(join(flow_dir_path, file_name))
         and not file_name.startswith(".")
-        and not "raise_error_flow" in file_name
-        and not "accelerator_flow" in file_name
+        and not file_name in non_standard_test_flows
     ]
     return file_paths
 
 
+def test_s3_sensor_flow(pytestconfig) -> None:
+    # ensure the s3_sensor waits for some time before the key exists
+    file_name = f"{uuid.uuid1()}.txt"
```

Comment:
the file name should start with its purpose rather than a uiid only to be self documenting.